zerver/management/commands/deactivate_realm.py

```diff
@@ -9,15 +9,32 @@ class Command(ZulipBaseCommand):
     help = """Script to deactivate a realm."""
 
     def add_arguments(self, parser: ArgumentParser) -> None:
+        parser.add_argument('--redirect_url', metavar='<redirect_url>',
+                            help="URL to which the realm has moved")
         self.add_realm_args(parser, True)
 
     def handle(self, *args: Any, **options: str) -> None:
         realm = self.get_realm(options)
+
         assert realm is not None  # Should be ensured by parser
 
         if realm.deactivated:
             print("The realm", options["realm_id"], "is already deactivated.")
+            if ((not realm.deactivated_redirect) and (options['redirect_url'])):
```

Comment:
If we do want to allow setting the redirect URL for an already deactivated realm, and allow changing it if it was previously set, you previous approach of having a separate `do_add_deactivate_redirect` function may work better.

Reply:
Also, if we are allowing setting the redirect URL for a previously deactivated realm, it may be worth allowing changing it if it was previously set too.

Reply:
Yes makes sense! will make the change to allow setting redirect URL everytime the user uses the --redirect_url flag.